fastlane_bot/modes/triangle_bancor_v3_two_hop.py

```diff
@@ -8,362 +8,54 @@
-        """
-        see base.py
-        """
-        if self.base_exchange != "bancor_v3":
-            self.ConfigObj.logger.warning(
-                f"base_exchange must be bancor_v3 for {self.arb_mode}, setting it to bancor_v3"
-            )
-            self.base_exchange = "bancor_v3"
-
-        self.ConfigObj.logger.info(
-            f"flashloan_tokens for arb_mode={self.arb_mode} will be overwritten. "
-        )
-
-        self._check_limit_flashloan_tokens_for_bancor3()
-
-        if candidates is None:
-            candidates = []
-
-        # Get combinations of flashloan tokens
-        combos = self.get_combos(
-            self.flashloan_tokens, self.CCm, arb_mode=self.arb_mode
-        )
-
-        # Get the miniverse combinations
-        all_miniverses = self.get_miniverse_combos(combos)
-
-        if len(all_miniverses) == 0:
-            return None
-
-        # Check each source token and miniverse combination
-        for src_token, miniverse in all_miniverses:
-            r = None
-
-            try:
-                # Run main flow with the new set of curves
-                (
-                    profit_src,
-                    trade_instructions,
-                    trade_instructions_df,
-                    trade_instructions_dic,
-                ) = self.run_main_flow(miniverse, src_token)
-
-            except Exception:
-                continue
-            if trade_instructions_dic is None:
+    def get_combos(self) -> List[Any]:
+        fltkns = self.CCm.byparams(exchange="bancor_v3").tknys()
+        if self.ConfigObj.LIMIT_BANCOR3_FLASHLOAN_TOKENS:
+            # Filter out tokens that are not in the existing flashloan_tokens list
+            flashloan_tokens = [tkn for tkn in fltkns if tkn in self.flashloan_tokens]
+            self.ConfigObj.logger.info(f"limiting flashloan_tokens to {self.flashloan_tokens}")
+        else:
+            flashloan_tokens = fltkns
+
+        miniverse_combos = []
+
+        for tkn0, tkn1 in [(tkn0, tkn1) for tkn0, tkn1 in product(flashloan_tokens, flashloan_tokens) if tkn0 != tkn1]:
+            all_curves = list(set(self.CCm.bypairs(f"{tkn0}/{tkn1}")) | set(self.CCm.bypairs(f"{tkn1}/{tkn0}")))
+
+            carbon_curves = [curve for curve in all_curves if curve.params.get("exchange") in self.ConfigObj.CARBON_V1_FORKS]
+            if len(carbon_curves) == 0:
                 continue
-            if len(trade_instructions_dic) < 3:
-                continue
-            # Get the candidate ids
-            cids = [ti["cid"] for ti in trade_instructions_dic]
-
-            # Calculate the profit
-            profit = self.calculate_profit(src_token, profit_src, self.CCm, cids)
 
-            if str(profit) == "nan":
-                self.ConfigObj.logger.debug("profit is nan, skipping")
+            external_curves = [curve for curve in all_curves if curve.params.get("exchange") not in self.ConfigObj.CARBON_V1_FORKS]
+            if len(external_curves) == 0:
```

Comment:
This check and the check for carbon_curves above should be combined to:
```
if len(external_curves) == 0 and len(carbon_curves) == 0:
    continue
```

Reply:
If you meant `or`, then the second part of this check is already applied a few lines above.
If you meant `and` then please clarify the reason.

Reply:
We only want to `continue` if there are no curves to pair with the bancor v3 curves. 

If there are external curves or carbon curves, we want to complete the process.

Reply:
Revised, can you please take a look?